module/modmanager/manager.go

```diff
@@ -130,14 +139,21 @@ func (mgr *Manager) add(ctx context.Context, conf config.Module, conn *grpc.Clie
 		resources: map[resource.Name]*addedResource{},
 	}
 
+	// add calls startProcess, which can also be called by the OUE handler in the attemptRestart
+	// call. Both of these involve owning a lock, so in unhappy cases of malformed modules
+	// this can lead to a deadlock. To prevent this, we set inStartup here to indicate to
+	// the OUE handler that it shouldn't act while add is still processing.
+	mod.inStartup.Store(true)
+	defer mod.inStartup.Store(false)
```

Comment:
do we also need the lock here?

Reply:
I think we want to leave the `inStartupLock` alone so that the OUE handler can take it and return quickly. Otherwise, we'll still end up with the OUE handler hanging on trying to get the lock until `add` finishes.

Reply:
^ yeah we don't want `add` to grab that lock here. If it did, then the OUE will succeed in getting the lock after `add` finishes, find that `inStartup` is now false, and attempt to restart the module that might have started successfully already.

Reply:
cool, just slightly confused about the var name but it's been changed